src/System.Private.CoreLib/src/System/Runtime/CompilerServices/TaskAwaiter.CoreCLR.cs

```diff
@@ -0,0 +1,141 @@
+                etwLog.TaskWaitBegin(
+                    (currentTaskAtBegin != null ? currentTaskAtBegin.m_taskScheduler.Id : TaskScheduler.Default.Id),
+                    (currentTaskAtBegin != null ? currentTaskAtBegin.Id : 0),
+                    task.Id, TplEtwProvider.TaskWaitBehavior.Asynchronous,
+                    (continuationTask != null ? continuationTask.Id : 0));
+            }
+
+            // Create a continuation action that outputs the end event and then invokes the user
+            // provided delegate.  This incurs the allocations for the closure/delegate, but only if the event
+            // is enabled, and in doing so it allows us to pass the awaited task's information into the end event
+            // in a purely pay-for-play manner (the alternatively would be to increase the size of TaskAwaiter
+            // just for this ETW purpose, not pay-for-play, since GetResult would need to know whether a real yield occurred).
+            return AsyncMethodBuilderCore.CreateContinuationWrapper(continuation, (innerContinuation,innerTask) =>
+            {
+                if (Task.s_asyncDebuggingEnabled)
+                {
+                    Task.RemoveFromActiveTasks(innerTask.Id);
+                }
+
+                TplEtwProvider innerEtwLog = TplEtwProvider.Log;
+
+                // ETW event for Task Wait End.
+                Guid prevActivityId = new Guid();
+                bool bEtwLogEnabled = innerEtwLog.IsEnabled();
+                if (bEtwLogEnabled)
+                {
+                    var currentTaskAtEnd = Task.InternalCurrent;
+                    innerEtwLog.TaskWaitEnd(
+                        (currentTaskAtEnd != null ? currentTaskAtEnd.m_taskScheduler.Id : TaskScheduler.Default.Id),
+                        (currentTaskAtEnd != null ? currentTaskAtEnd.Id : 0),
+                        innerTask.Id);
+
+                    // Ensure the continuation runs under the activity ID of the task that completed for the
+                    // case the antecendent is a promise (in the other cases this is already the case).
+                    if (innerEtwLog.TasksSetActivityIds && (innerTask.Options & (TaskCreationOptions)InternalTaskOptions.PromiseTask) != 0)
+                        EventSource.SetCurrentThreadActivityId(TplEtwProvider.CreateGuidForTaskID(innerTask.Id), out prevActivityId);
+                }
+
+                // Invoke the original continuation provided to OnCompleted.
+                innerContinuation();
+
+                if (bEtwLogEnabled)
+                {
+                    innerEtwLog.TaskWaitContinuationComplete(innerTask.Id);
+                    if (innerEtwLog.TasksSetActivityIds && (innerTask.Options & (TaskCreationOptions)InternalTaskOptions.PromiseTask) != 0)
+                        EventSource.SetCurrentThreadActivityId(prevActivityId);
+                }
+            }, task);
+        }
+    }
+}
```

Comment:
I assume this whole file is just a straight move of the source shown as deleted earlier from TaskAwaiter.cs and nothing in it actually changed that needs to be reviewed?

Reply:
I expect that this split won't be necessary as we unify the rest of the Async/Task implementation. If it is the case, handling the difference temporarily using ifdefs would be better.

Reply:
Correct. I made no implementation change, just a partial move to shared. 

This diff (which reduces diff in the file) shows how corert version changed to result both repos in the same version of file: https://github.com/dotnet/corert/pull/5893/commits/7ca6a844024611a861337b99f0bac2a9241d6d6c